test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -347,6 +350,31 @@ spec =
             fill 10 # no change
             atom.bar . should_equal 2
 
+        Test.specify "construct and fill with type" <|
+            pair = Meta.atom_with_hole (e -> My_Ascribed_Type.Value_With_Type 1 e 3)
+
+            atom = pair.value
+            fill = pair.fill
+
+            Meta.is_atom atom . should_be_true
+
+            atom.foo . should_equal 1
+            atom.baz . should_equal 3
+            case atom.bar of
+                n : Number -> Test.fail "Shouldn't be number yet: "+n
+                _ -> Nothing
+
+            tpe_of_bar = Meta.type_of atom.bar
+            tpe_of_bar.is_error . should_equal True
+
+            fill 2
+            atom.bar . should_equal 2
+            tpe_of_bar_after_fill = Meta.type_of atom.bar
+            tpe_of_bar_after_fill . should_equal Integer
```

Comment:
Can we add a test case where we use `fill "FOO"`?

Will the typecheck then be ignored or will it check it when filling the hole?

Reply:
Good idea. If I replace `fill 2` with `fill "Two"`, then the `fill` function finishes and then the test fails with:
```
    - [FAILED] construct and fill with type [30ms]
        Reason: 'Two' did not equal 2 (at enso/test/Tests/src/Semantic/Meta_Spec.enso:371:13-37).
```
either I find a simple way to fix it or I report another issue.

Reply:
Thanks, I think it would be good to get it fixed.

I imagine having a type ascription that is actually not checked is not good. In such case I'd rather not allow type ascriptions where holes are expected - let's clearly mark that if you need a hole, you cannot have type checking and so need to not provide a type ascription.

Reply:
Got it working in a6f3b0f, but I will have to polish it a bit.